core/gui/src/app/workspace/component/left-panel/left-panel.component.ts

```diff
@@ -95,4 +100,37 @@ export class LeftPanelComponent implements OnDestroy, OnInit {
       this.height = height!;
     });
   }
+
+  resetPanelPosition() {
+    this.dragPosition = { x: this.returnPosition.x, y: this.returnPosition.y };
+  }
+
+  parseTranslate3d(translate3d: string): [number, number, number] {
+    const regex = /translate3d\((-?\d+\.?\d*)px,\s*(-?\d+\.?\d*)px,\s*(-?\d+\.?\d*)px\)/g;
+    const match = regex.exec(translate3d);
+    if (match) {
+      const x = parseFloat(match[1]);
+      const y = parseFloat(match[2]);
+      const z = parseFloat(match[3]);
+      return [x, y, z];
+    }
+    return [0, 0, 0];
+  }
+
+  calculateTotalTranslate3d(translates: string): [number, number, number] {
```

Comment:
Can we put this function in a separate file instead of duplicate this function 3 times?